subcommands/targets/offline-update.go

```diff
@@ -132,8 +134,13 @@ Notice that multiple targets in the same directory is only supported in LmP >= v
 				subcommands.DieNotNil(err)
 			}
 
-			fmt.Printf("Downloading Apps fetched by the `assemble-system-image` run; build number:  %d, tag: %s...\n", ti.version, ti.buildTag)
-			err = downloadApps(factory, targetName, ti.version, ti.buildTag, path.Join(dstDir, "apps"))
+			if ti.fetchedApps == nil {
+				fmt.Printf("Downloading Apps fetched by the `assemble-system-image` run; build number: %d, tag: %s...\n", ti.version, ti.buildTag)
+				err = downloadApps(factory, targetName, ti.version, ti.buildTag, path.Join(dstDir, "apps"))
+			} else {
+				fmt.Printf("Downloading Apps fetched by the `publish-compose-apps` run; apps: %s, uri: %s...\n", ti.fetchedApps.Shortlist, ti.fetchedApps.Uri)
+				err = downloadAppsArchive(ti.fetchedApps.Uri, path.Join(dstDir, "apps"))
+			}
```

Comment:
> I guess 1 year from now or so we should be able to remove legacy handling from here, right?
> Is that your anticipation?

Yes.

Reply:
> So, this is another optimization, which becomes possible because of recent changes to ci-scripts, right?

It's not exactly optimization, this is how it should work for the customers that enables offline update without enabling preloading.